### PR TITLE
Tons of Bridge Tests

### PR DESCRIPTION
**Pull request by prestwich**
tracking PR for ongoing test work

### CODE REVIEW

File: typescript/optics-tests/test/bridge/bridge.test.ts [hunk 1]
```diff
@@ -17,7 +21,7 @@ const BRIDGE_MESSAGE_TYPES = {
 
 const typeToBytes = (type: number) => `0x0${type}`;
 
-describe('Bridge', async () => {
+describe.only('BridgeRouter', async () => {
```
Comment (anna-carroll) [inline typescript/optics-tests/test/bridge/bridge.test.ts:24]: remove `.only`

File: typescript/optics-tests/test/bridge/bridge.test.ts [hunk 2]
```diff
@@ -57,176 +60,203 @@ describe('Bridge', async () => {
...
-        deployerId,
-      );
+        const representation = await deploy.getTestRepresentation();
+        expect(representation).to.not.be.undefined;
+        repr = representation!;
 
-      await expect(stealTx).to.be.revertedWith(
-        'ERC20: burn amount exceeds balance',
-      );
+        await expect(handleTx).to.emit(deploy.bridgeRouter!, 'TokenDeployed');
```
Comment (anna-carroll) [inline typescript/optics-tests/test/bridge/bridge.test.ts:95]: okay i am slightly confused about how this passes - doesn't this execute `handleTx` twice? in which case, the tx on line 84 would emit the event and the tx on 95 would not?

this is passing tho so i guess it somehow runs the same tx

**Review by anna-carroll — COMMENTED** [review]
lookin good

File: typescript/optics-tests/test/bridge/bridge.test.ts (hunk 1 above)
Comment (prestwich) [reply typescript/optics-tests/test/bridge/bridge.test.ts:24]: after it's ready for merge

**Comment by prestwich** [comment]
two things to document better
1. MockCore needs to consume a lot of gas to simuilate being home
2. The test `PROCESS_GAS`  in `bridge.test.ts` needs to be set LOWER than the Replica `PROCESS_GAS` to ensure that we have a safety margin

**Review by anna-carroll — APPROVED** [review]
Amazing!~!

**Pull request merged**